picard/ui/metadatabox.py

```diff
@@ -309,16 +309,29 @@ def contextMenuEvent(self, event):
             if len(tags) == 1 or removals or useorigs:
                 menu.addSeparator()
             menu.addAction(self.add_tag_action)
+            add_to_preserved_tags_action = QtGui.QAction(_(u"Add Selected Tags to 'Preserve Tags' List"), self.parent)
+            add_to_preserved_tags_action.triggered.connect(self.add_to_preserved_tags)
+            menu.addAction(add_to_preserved_tags_action)
             menu.addSeparator()
         menu.addAction(self.changes_first_action)
         menu.exec_(event.globalPos())
         event.accept()
 
+    def add_to_preserved_tags(self):
+        # Get all selected tags including '~length'
+        tags = self.selected_tags()
+        preserved_tags = [tag for tag in
+                          map(lambda x: x.strip(), config.setting['preserved_tags'].split(','))
+                          if tag != ""]
+        preserved_tags.extend(tags)
+        config.setting['preserved_tags'] = ", ".join(filter(lambda x: x in TAG_NAMES.keys(),
```

Comment:
Unfortunately, `TAG_NAMES` can't be used like this because it's not a complete list of tag names that can be written (such a list would be impossible to make with formats where tag names can be anything).

Additionally, enabling the content menu action and then silently not adding the tag to the list is not good.